clarify build on docs content and move doc to dag section (#5633)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/documentation.md

---
title: "Documentation"
description: "Learn how good documentation for your dbt models helps stakeholders discover and understand your datasets."
id: "documentation"
---

> Good documentation for your dbt models will help downstream consumers discover and understand the datasets which you curate for them.

Reviewer: ```suggestion
Good documentation for your dbt models will help downstream consumers discover and understand the datasets you curate for them.
```

dbt provides a way to generate documentation for your dbt project and render it as a website.

> ## Related documentation

Reviewer: Should we move this to the bottom while we're here, since that's where we usually place it?

Author: tbh i did think that but the other 'build' docs have this format so left it so it would be consistent for users.

## Overview

> The default documentation experience in dbt Cloud is [dbt Explorer](/docs/collaborate/explore-projects), available on [Team or Enterprise plans](https://www.getdbt.com/pricing/), which provides a richer and more interactive experience for understanding your project's resources and lineage. dbt Cloud developer and dbt Core users can use [dbt Docs](/docs/collaborate/legacy-dbt-docs), a legacy feature to generate documentation, and is accessible through dbt Explorer or the **Documentation** link.

Author: added explorer content

</File>

## Generating project documentation

The default documentation experience in dbt Cloud is [dbt Explorer](/docs/collaborate/explore-projects), available on [Team or Enterprise plans](https://www.getdbt.com/pricing/). Use dbt Explorer to view your project's resources (such as models, tests, and metrics), its [metadata](/docs/collaborate/explore-projects#generate-metadata), and their lineage to gain a better understanding of its latest production state.

dbt Cloud developer and dbt Core users can use [dbt Docs](/docs/collaborate/build-and-view-your-docs#dbt-docs), a legacy feature that generates basic documentation, but it doesn't offer the same speed, metadata, or visibility as dbt Explorer. 

Generate documentation for you project by following these steps:

> - First, run `dbt docs generate` — this command tells dbt to compile relevant information about your dbt project and warehouse into `manifest.json` and `catalog.json` files respectively. 

Reviewer: Since we're giving them steps to follow in order, suggest a numbered list
```suggestion
1. Run `dbt docs generate` — this command tells dbt to compile relevant information about your dbt project and warehouse into `manifest.json` and `catalog.json` files, respectively. 
```

> - To view the documentation for all columns and not just columns described in your project, ensure that you have created the models with `dbt run` beforehand.

Reviewer: ```suggestion
2. Ensure that you have created the models with `dbt run` to view the documentation for all columns, not just those described in your project.
```

> - If you're developing locally, run `dbt docs serve` to use these `.json` files to populate a local website.

Reviewer: ```suggestion
3. Run `dbt docs serve` if you're developing locally to use these `.json` files to populate a local website.
```

</File>

## Generating project documentation

> The default documentation experience in dbt Cloud is [dbt Explorer](/docs/collaborate/explore-projects), available on [Team or Enterprise plans](https://www.getdbt.com/pricing/). Use dbt Explorer to view your project's resources (such as models, tests, and metrics), its [metadata](/docs/collaborate/explore-projects#generate-metadata), and their lineage to gain a better understanding of its latest production state.

Author: @cafzal introducing explorer content here - what do you think?

also moved this page from collaborate -> build your dag. however, there is also an option to have a 'doc your project' folder under the 'build' section. and add doc examples for other resources per Dave's suggestion.

Reviewer 2: sounds good!

> dbt Cloud developer and dbt Core users can use [dbt Docs](/docs/collaborate/build-and-view-your-docs#dbt-docs), a legacy feature that generates basic documentation, but it doesn't offer the same speed, metadata, or visibility as dbt Explorer. dbt Docs is available to dbt Cloud developer plans or dbt Core users.

Author: ```suggestion
dbt Cloud developer and dbt Core users can use [dbt Docs](/docs/collaborate/build-and-view-your-docs#dbt-docs), a legacy feature that generates basic documentation, but it doesn't offer the same speed, metadata, or visibility as dbt Explorer. 
```

</File>

## Generating project documentation

> You can generate a documentation site for your project (with or without descriptions) in development using your editor or with [dbt Explorer](/docs/collaborate/explore-projects) in dbt Cloud after a successful job run in the deployment environment.

Reviewer 2: @mirnawong1 i'd frame this as 'you can use explorer to view the latest metadata about your project, or generate a static docs site'. then point the reader to the explorer docs to read about setup (within the generate metadata section). no need to link to the docs job article for explorer setup. generating docs is still an option for folks who don't have explorer or want more flexibility, but shouldn't be our default suggestion.

In the resulting documentation, `'{{ doc("table_events") }}'` will be expanded to the markdown defined in the `table_events` docs block.

## Setting a custom overview

> *Currently available for the legacy dbt Docs only.*

Reviewer: Should we be referring to it as `legacy` since it's still the docs experience for Core users? They'll be reading this page too, right? Maybe the "Previous dbt Cloud docs experience" or something similar?

In the resulting documentation, `'{{ doc("table_events") }}'` will be expanded to the markdown defined in the `table_events` docs block.

### Setting a custom overview

> *Currently available for the legacy dbt Docs only.*

Author: adding this clarification. opted for this small italics rather than separating this with a header since e some of these configs will be available in explorer.

</File>

### Navigating the documentation site

> Use [dbt Explorer](/docs/collaborate/explore-projects) for a richer documentation experience and more interactive experience for understanding your project's resources and lineage. Available on [Team or Enterprise plans](https://www.getdbt.com/pricing/).

Author: @cafzal introducing explorer content here - what do you think?

<Lightbox src="/img/docs/building-a-dbt-project/testing-and-documentation/ec77c45-Screen_Shot_2018-08-14_at_6.31.56_PM.png" title="Opening the DAG mini-map"/>

In this example, the `fct_subscription_transactions` model only has one direct parent. By clicking the "Expand" button in the top-right corner of the window, we can pivot the graph horizontally and view the full <Term id="data-lineage">lineage</Term> for our model. This lineage is filterable using the `--select` and `--exclude` flags, which are consistent with the semantics of [model selection syntax](/reference/node-selection/syntax). Further, you can right-click to interact with the DAG, jump to documentation, or share links to your graph visualization with your coworkers.

<Lightbox src="/img/docs/building-a-dbt-project/testing-and-documentation/ac97fba-Screen_Shot_2018-08-14_at_6.35.14_PM.png" title="The full lineage for a dbt model"/>

</Expandable>

### Deploying the documentation site

> With dbt Cloud, use [dbt Explorer](/docs/collaborate/explore-projects) automatically retrieves the metadata updates after each job run in the production or staging deployment environment so it always has the latest results for your project &mdash; meaning it's always automatically updated after each job run.

Author: @cafzal introducing explorer content here - what do you think?

<Lightbox src="/img/docs/building-a-dbt-project/testing-and-documentation/ac97fba-Screen_Shot_2018-08-14_at_6.35.14_PM.png" title="The full lineage for a dbt model"/>

</Expandable>

## Deploying the documentation site

With dbt Cloud, use [dbt Explorer](/docs/collaborate/explore-projects) automatically retrieves the metadata updates after each job run in the production or staging deployment environment so it always has the latest results for your project &mdash; meaning it's always automatically updated after each job run.

:::caution Security

The `dbt docs serve` command is only intended for local/development hosting of the documentation site. Please use one of the methods listed below (or similar) to ensure that your documentation site is hosted securely!

:::

#### For dbt Docs users

dbt's documentation website was built to make it easy to host on the web. The site is "static,” meaning you don't need any "dynamic" servers to serve the docs. You can host your documentation in several ways:

> * Use [dbt Cloud](/docs/collaborate/build-and-view-your-docs)'s default documentation experience with [dbt Explorer](/docs/collaborate/explore-projects).

Reviewer: ```suggestion
* Use [dbt Cloud's](/docs/collaborate/build-and-view-your-docs) default documentation experience with [dbt Explorer](/docs/collaborate/explore-projects).
```
